src/ai/susi/server/api/aaa/PasswordChangeService.java

```diff
@@ -62,55 +62,62 @@ public ServiceResponse serviceImpl(Query post, HttpServletResponse response, Aut
         String password = post.get("password", null);
         String newpassword = post.get("newpassword",null);
 
-
-        ClientCredential pwcredential = new ClientCredential(ClientCredential.Type.passwd_login, useremail);
-        Authentication authentication = DAO.getAuthentication(pwcredential);
-        ClientCredential emailcred = new ClientCredential(ClientCredential.Type.passwd_login,
-                authentication.getIdentity().getName());
-        ClientIdentity identity = authentication.getIdentity();
-        String passwordHash;
-        String salt;
-
-        try {
-            passwordHash = authentication.getString("passwordHash");
-            salt = authentication.getString("salt");
-        } catch (Throwable e) {
-            Log.getLog().info("Invalid password try for user: " + identity.getName() + " from host: " + post.getClientHost() + " : password or salt missing in database");
-            result.put("message", "invalid credentials");
-            throw new APIException(422, "Invalid credentials");
+        if(password.equals(newpassword)){
+            result.put("message", "Your current password and new password matches");
```

Comment:
This is not the best approach because an error should always be sent as an error. But since catching and showing error in ReactJS is not possible, This is the next best approach. So ya.. go on with this for now. Also, the error message could be more informative like "Current password and New password can not be similar".